tests/RobotFramework/tests/tedge_watchdog.robot

```diff
@@ -0,0 +1,115 @@
+*** Settings ***
+Documentation
+...                 Test suite for tedge-watchdog service.
+...
+...                 Reference: docs/src/operate/monitoring/systemd-watchdog.md
+
+Resource            ../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Custom Teardown
+
+Test Tags           adapter:docker    theme:tedge-watchdog
+
+
+*** Variables ***
+# Interval at which we send notifications/check health of the service. Use smaller value for test to complete faster but
+# increase if flaky.
+${WATCHDOG_SEC}     5
+${SERVICE_NAME}     tedge-agent
+
+
+*** Test Cases ***
+# NOTE: This test succeeds, but not for the reason we expect - instead of simply not sending a notification, upon
+# reciving a status:down message tedge-watchdog exits for unknown reason, so it can't send the notification and the
+# monitored service is restarted by systemd, along with tedge-watchdog itself.
+# This is a result of a specific configuration of these services' unit files, that we're not checking or validating
+# anywhere, so we need to define this behaviour more precisely by providing additional tests.
+Watchdog doesn't send notification if service is down which leads to service restart
+    Start Service    ${SERVICE_NAME}
+    Start Service    tedge-watchdog
+
+    # wait for service to start
+    ${pid} =    Service Should Be Running    ${SERVICE_NAME}
+
+    # Simulate service signalling that it's unhealthy
+    ${before_restart_timestamp} =    Get Unix Timestamp
+    Execute Command
+    ...    tedge mqtt pub 'te/device/main/service/${SERVICE_NAME}/status/health' '{"status": "down", "pid": ${pid}, "time": "${before_restart_timestamp}"}'
```

Comment:
This may not be fully deterministic as the watchdog would be waiting for a response **after** it sends the health check (with a later timestamp) and just ignores other random health messages that it receives with a lower timestamp. So, we need to make sure that this response reaches the watchdog after the health check request is received, with a higher timestamp, and also ensure that it reaches before the real running `tedge-agent` responds with its `up` status.

Reply:
An way to *disconnect* tedge-watchdog from the MQTT topics used by the agent and the mapper (for testing purposes) is to configure tedge-watchdog with a fake topic root (say `test-te`).